documentation/contributing/pullrequests.markdown

```diff
@@ -0,0 +1,85 @@
+
+### keep requests clean and independent
+
+Different commits in a request should correspond to logical units that build up on each other. Feel free to rebase and force-push your feature-branch to address feedback.
+If your request consists of multiple commits with titles like "cleanup braces" or "address xyz's feedback", the maintainer who merges the request will squash-merge and your changes will end up as a single commit in the project.
+
+Logically independent changes should be separated into multiple pull-requests. E.g. don't create a pull-request that changes plan execution and planning scene updates at the same time if there is no direct connection between the two changesets. If there *is* a connection that is not immediately obvious, please explain it in your request. "I had to change it to get my setup to work" is no connection. You might experience separate independent issues.
+This makes it much more likely to get your changes merged soon, because if maintainers are unsure about one change, they can still merge the other one after review.
+
+### ping
+
+If you feel everyone forgot about your request, add a short note saying "ping" or "Could someone please review/merge this?" to the request. It might be that the item dropped of everyone's TODO list for no reason.
+
+### review requests and +1
+
+All feedback on existing pull-requests is welcome and appreciated. If you are familiar with the code someone requests to change, please step up and review their requests. The maintainer's time is often quite limited. This usually boils down to answering the following questions and provide feedback in the request:
+
+ - Does the patch do what the author claims it does?
+ - Is it general enough to work with everyones use-case and not only the requestor's setup?
+ - Does it address all edge-cases you are aware of?
+ - If a new interface is introduced: Is this interface general enough for likely future modifications? Otherwise we might have to break API later on if we wish to add more useful features to the same module.
+ - Is the change [ABI compatible](https://en.wikipedia.org/wiki/Application_binary_interface) or is there a good reason (e.g. safety) to break it?
+ - Does the patch respect the [MoveIt! Code Style](code.html)?
+
+If (and only if) you answer all these questions with "yes", add a note saying "+1" to the request. This does **not** mean you like to have the claimed feature in MoveIt!. If it's not the first message you ever sent on a public channel concerning MoveIt, it means the next maintainer who reviews this request is free to merge it after review.
+If you are unsure about some part of the request, feel free to add a note and ask the author for details and explanations of individual changes.
+
+## Maintainer Side
+
+The ``*-devel`` branches of the repositories must always compile on the target platforms.
+
+The ``*-devel`` branches must not be force-pushed.
```

Comment:
On Wed, Aug 03, 2016 at 12:28:35PM -0700, Dave Coleman wrote:
> *should never be force-pushed

I intentionally put "must not" here because the problematic cases are always the "well, but, you see, in this case, _should_ is not too important..."

> Side note: I'm planning on disabling force-pushing within the repo's Github settings after the switch

+1 to disable force-push on the `*-devel` branches. (only on them though)